src/hpr_routing.erl

```diff
@@ -65,22 +77,25 @@ dispatch_packet({join_req, {AppEUI, DevEUI}}, Packet) ->
         ],
         "handling join"
     ),
-    {deliver_packet(Packet, Routes), erlang:length(Routes)};
+    RoutingInfo = {eui, DevEUI, AppEUI},
+    {deliver_packet(Packet, Routes, RoutingInfo), erlang:length(Routes)};
 dispatch_packet({uplink, DevAddr}, Packet) ->
     lager:debug(
         [{devaddr, hpr_utils:int_to_hex(DevAddr)}],
         "handling uplink"
     ),
     Routes = hpr_config:lookup_devaddr(DevAddr),
-    {deliver_packet(Packet, Routes), erlang:length(Routes)}.
+    RoutingInfo = {devaddr, DevAddr},
+    {deliver_packet(Packet, Routes, RoutingInfo), erlang:length(Routes)}.
```

Comment:
Alternative to consider. In `deliver_packet`, count the number of routes and the number of successfully delivered routes? Possibly broken down by type?

Reply:
The number of routes is for metric purposes. 
We don't currently care about routes by type. If we do in the future we can change it then.